src/main/java/uk/gov/hmcts/ccd/data/casedetails/search/SearchQueryFactoryOperation.java

```diff
@@ -99,16 +98,13 @@ private String addUserCaseStateAccessClause(MetaData metadata, Map<String, Objec
     }
 
     private void addParameters(final Query query, List<Criterion> criterion) {
-
-        IntStream.range(0, criterion.size())
-                .forEach(position -> query.setParameter(position, criterion.get(position).getSoughtValue()));
+        criterion.forEach(criteria -> query.setParameter(criteria.buildParameterId(), criteria.getSoughtValue()));
     }
 
     private String toClauses(final List<Criterion> criterion) {
-        return IntStream.range(0, criterion.size())
-                .mapToObj(Integer::new)
-                .map(position -> criterion.get(position).buildClauseString(position, getOperation()))
-                .collect(Collectors.joining(AND));
+        return criterion.stream()
+            .map(criteria -> criteria.buildClauseString(getOperation()))
+            .collect(Collectors.joining(AND));
     }
```

Comment:
Functional change that will require reviewing. With Java 11-compatible version of Hibernate we can no longer use the old approach of mixing named and positional parameters in the same query.